m3/src/main/java/com/uber/m3/tally/m3/M3Reporter.java

```diff
@@ -484,46 +494,46 @@ public void run() {
                     // catch block.
                     SizedMetric sizedMetric = metricQueue.poll(maxBufferingDelay.toMillis(), TimeUnit.MILLISECONDS);
 
-                    // Drop metrics that came in after close
-                    if (sizedMetric == SizedMetric.CLOSE) {
-                        metricQueue.clear();
-                        break;
-                    }
-
                     if (sizedMetric == null) {
                         // If we didn't get any new metrics after waiting the specified time,
                         // flush what we have so far.
-                        process(SizedMetric.FLUSH);
+                        flushBuffered();
                     } else {
                         process(sizedMetric);
                     }
-
+                } catch (Throwable t) {
+                    // This is fly-away guard making sure that uncaught exception
+                    // will not crash the processor
+                    LOG.error("Unhandled exception in processor", t);
                 }
-            } catch (InterruptedException e) {
-                // Don't care if we get interrupted - the finally block will clean up
-            } finally {
-                drainQueue();
-                flushBuffered();
-                // Count down shutdown latch to notify reporter
-                shutdownLatch.countDown();
             }
+
+            LOG.warn("Processor shutting down");
```

Comment:
should this be debug?

Reply:
This should not be debug, to make sure it surfaces in the logs (debug is usually disasbled)

Reply:
That was the intention behind the question - does it need to be in logs? if you want them, I don't have problem with it.